src/widgetastic_patternfly4/bulletchart.py

```diff
@@ -0,0 +1,166 @@
+import re
+
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import Text
+from widgetastic.widget import View
+from widgetastic.xpath import quote
+
+
+class Legend:
+    """Represents Legend of chart."""
+
+    def __init__(self, label, value=None, color=None, element=None):
+        self.label = label
+        self.value = value
+        self.color = color
+        self.element = element
+
+    def click(self):
+        self.element.click()
+
+    def __repr__(self):
+        return f"Legend({self.label}: {self.value})"
+
+    def __gt__(self, leg):
+        return self.__class__ == leg.__class__ and self.value > leg.value
+
+    def __eq__(self, leg):
+        return (
+            self.__class__ == self.__class__ and self.label == leg.label and self.value == leg.value
+        )
+
+
+class DataPoint(Legend):
+    """Represents DataPoint on chart."""
+
+    def __repr__(self):
+        return f"DataPoint({self.label}: {self.value})"
+
+
+class BulletChart(View):
+    """Represents the Patternfly Bullet Chart.
+
+    https://www.patternfly.org/v4/documentation/react/charts/chartbullet
+
+    Args:
+        id: If you want to look the input up by id, use this parameter, pass the id.
+        locator: If you have specific locator else it will take pf-chart.
+        offset_denominator: Denominator for offset value calculation.
+    """
+
+    ROOT = ParametrizedLocator("{@locator}")
+    DEFAULT_LOCATOR = ".//div[contains(@class, 'chartbullet')]"
+    LEGEND_ICON = (
+        ".//*[name()='g' or name()='svg']/*[name()='rect']/following-sibling::*[name()='path']"
+    )
+    LEGEND_TEXT = ".//*[name()='text' and contains(@id, 'legend-labels')]"
+    LEGEND_ITEM_REGEX = re.compile(r"(\d+)\s(\w.*)|(\w.*)\s(\d+)")
+
+    ITEMS = ".//*[name()='g']/*[name()='path' and not(contains(@style, 'type:square'))]"
+    TOOLTIP_REGEX = re.compile(r"(.*?): ([\d]+)")
+    APPLY_OFFSET = True
+
+    tooltip = Text(
+        ".//*[name()='svg' and contains(@aria-labelledby, 'victory-container')]/"
+        "following-sibling::div[contains(@style, 'z-index')]/*[name()='svg']"
+    )
+
+    def __init__(self, parent=None, id=None, locator=None, logger=None, *args, **kwargs):
+        View.__init__(self, parent=parent, logger=logger)
+        if id:
+            self.locator = ".//div[@id={}]".format(quote(id))
+        elif locator:
+            self.locator = locator
+        else:
+            self.locator = self.DEFAULT_LOCATOR
+
+        self.args = args
+        self.kwargs = kwargs
+
+    def _offsets(self, el):
+        """Calculate offset. Need to set offset with try and error method."""
+        offset_denominator = self.kwargs.pop("offset_denominator", 2.5)
+        size = self.browser.size_of(el)
+        width = size.width  # el.size["width"]
+        height = size.height  # el.size["height"]
```

Comment:
Why do we need these comments?

Reply:
my bad I was doing experiment and push as it its.